WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsHubTimeRangeController.swift

```diff
@@ -0,0 +1,146 @@
+import Foundation
+import Yosemite
+
+struct TimeRange {
+    let start: Date
+    let end: Date
+}
+
+public class AnalyticsHubTimeRangeController {
+
+    private let currentTimezone = TimeZone.current
+    private let currentDate: Date
+    private let currentCalendar: Calendar
+    private let selectionType: SelectionType
+
+    lazy private(set) var currentTimeRange: TimeRange = {
+        generateCurrentTimeRangeFrom(selectionType: selectionType)
+    }()
+
+    lazy private(set) var previousTimeRange: TimeRange = {
+        generatePreviousTimeRangeFrom(selectionType: selectionType)
+    }()
+
+    lazy private(set) var currentRangeDescription: String = {
+        generateDescriptionOf(timeRange: currentTimeRange)
+    }()
+
+    lazy private(set) var previousRangeDescription: String = {
+        generateDescriptionOf(timeRange: previousTimeRange)
+    }()
+
+    var selectionDescription: String {
+        selectionType.description
+    }
+
+    init(selectedTimeRange: StatsTimeRangeV4,
+         currentDate: Date = Date(),
+         currentCalendar: Calendar = Calendar(identifier: .iso8601)
+    ) {
```

Comment:
Style nit: I think we should have these braces in the previous line to match the rest of the method signatures style.

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/8193/commits/1778fc46df248a569ebdc7aaf809c3b639155d5d